src/WORKSPACE

```diff
@@ -48,17 +48,9 @@ http_archive(
     url = "https://bitbucket.org/eigen/eigen/get/c5e90d9.tar.gz",
 )
 
-http_archive(
+new_local_repository(
     name = "libusb",
     build_file = "@//external:libusb.BUILD",
-    sha256 = "a5fc7c3c8ecfaca3e149ffb5a422f358fc5276fe56d851f46a4023d1e061bc00",
-    strip_prefix = "libusb-1.0.23",
-    url = "https://github.com/libusb/libusb/archive/v1.0.23.zip",
-)
-
-new_local_repository(
-    name = "libudev",
-    build_file = "@//external:libudev.BUILD",
     path = "/",
 )
```

Comment:
I'm _really_ not a fan of just using the system libs. We should try to make our build as self-contained as possible. What was the reasoning behind this?

Reply:
Multiple reasons:
- Not all platforms provide libudev
- Simplifies things further by eliminating the need to download the libusb sources
- Not a fan of using system libs as well, but since we already did that for libudev might as well directly use precompiled libusb and eliminate the (explicit) libudev dependency

Reply:
*should clarify first point, not all platforms expose a userspace udev lib

Reply:
Hold up, that simplifies things _here_, but ideally we want to get to a point where we don't need to download any software dependencies in the setup script, and can do it all in bazel. Could we not do some platform-specific work for libudev (or the MacOS alternative)?

Reply:
So it turns out macOS doesn't support udev at all (part of systemd... which is Linux-only).  
I suspect `autoconf` in the libusb sources also configures the srcs for the specific platform being built, which is why directly including the sources in our build may or may not be portable; in which case I do prefer including binaries installed outside of our Bazel that are guaranteed to be configured properly for that platform.
I tried to configure and build libusb using genrules (I'm sure you've tried this already too) and it didn't really work out. Another alternative I'm considering is having a repository of prebuilt binaries (libusb and qt especially) that can be downloaded at build-time to work around this problem.

Reply:
Hmm, the _ideal_ would be using the options to build libusb stuff from source. Let's sync up today on this.

Reply:
I agree with that, might as well discuss Qt while we're at it. 
Going have to be a video call though, I'm WFH today

Reply:
After offline discussion, i'm OK with this.